---
title: "Déclencheurs d’Actions : objet API custom-phone-provider"
permalink: "action-triggers-custom-phone-provider-api-object"
'description': "Apprenez à utiliser l’objet API custom-phone-provider."
'og:title': "Déclencheurs d’Actions : objet API custom-phone-provider"
'og:description': "Apprenez à utiliser l’objet API custom-phone-provider."
'og:image': "https://cdn2.auth0.com/docs/1.14567.0/img/share-image.png"
'twitter:title': "Déclencheurs d’Actions : objet API custom-phone-provider"
'twitter:description': "Apprenez à utiliser l’objet API custom-phone-provider."
---

<Callout icon="file-lines" color="#0EA5E9" iconType="regular">

Les fournisseurs de messagerie téléphonique sont actuellement en accès anticipé. Pour en savoir plus sur les étapes de diffusion d’Auth0, consultez [Étapes de la diffusion du produit](https://auth0.com/docs/troubleshoot/product-lifecycle/product-release-stages).

</Callout>

L’objet API pour le déclencheur d’Actions `custom-phone-provider` comprend ce qui suit :

## `api.cache`

Stockez et récupérez des données qui persistent entre les exécutions.

### `api.cache.delete(key)`

Supprimez un enregistrement décrivant une valeur mise en cache sous la clé fournie, le cas échéant.

Renvoie un objet `CacheWriteResult` avec le type : `"success"` si une valeur a été supprimée du cache. Une opération qui a échoué renvoie le type : `"error"`. En cas d’erreur, l’objet renvoyé affichera une propriété code qui indique la nature de l’échec.

<table class="table">
<thead>
<tr>
<th>Paramètre</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>`key`</td>
<td>
<p>
<em>Chaîne</em>. La clé de l’enregistrement est stockée dans le cache.
            </p>
</td>
</tr>
</tbody>
</table>

### `api.cache.get(key)`

Récupérez un enregistrement décrivant une valeur mise en cache sous la clé, le cas échéant. Si un enregistrement est trouvé, la valeur mise en cache peut être trouvée dans la propriété valeur de l’objet retourné.

Renvoie un enregistrement si un élément est trouvé dans le cache sous la `key` fournie. Les enregistrements de cache sont des objets dotés d’une propriété `value` contenant la valeur mise en cache et d’une propriété `expires_at` indiquant l’heure d’expiration maximale de l’enregistrement, en millisecondes depuis l’époque Unix.

**Important** : Ce cache est conçu pour des données éphémères de courte durée. Certains éléments peuvent ne pas être disponibles lors de transactions ultérieures, même s’ils sont encore dans la limite de leur durée de vie.

<table class="table">
<thead>
<tr>
<th>Paramètre</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>`key`</td>
<td>
<p>
<em>Chaîne</em>. La clé de l’enregistrement est stockée dans le cache.
            </p>
</td>
</tr>
</tbody>
</table>

### `api.cache.set(key, value, [options])`

Stockez ou mettez à jour la valeur d’une chaîne dans le cache sous la clé spécifiée.

Les valeurs stockées dans ce cache sont limitées au déclencheur dans lequel elles sont définies. Elles sont soumises aux [Limites de cache des actions](https://auth0.com/docs/customize/actions/limitations).

Les valeurs stockées de cette manière auront des durées de vie allant jusqu’aux valeurs `ttl` ou `expires_at` spécifiées. Si aucune durée de vie n’est spécifiée, elle sera de 15 minutes par défaut. Les durées de vie ne doivent pas dépasser la durée maximale indiquée dans : [Limites de caches des actions](https://auth0.com/docs/customize/actions/limitations).

`options.ttl`

Numéro facultatif. Valeur de la durée de vie de cette entrée de cache en millisecondes. Bien que les valeurs mises en cache puissent être expulsés plus tôt, elles ne resteront jamais au-delà de la valeur `ttl` fournie.

Remarque : Cette valeur ne doit pas être fournie si une valeur a également été donnée pour `expires_at`. La première des deux échéances sera retenue si les deux options sont fournies.

<table class="table">
<thead>
<tr>
<th>Paramètre</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>`key`</td>
<td>
<p>
<em>Chaîne</em>. La clé de l’enregistrement est stockée dans le cache.
            </p>
</td>
</tr>
<tr>
<td>`value`</td>
<td>
<p>
<em>Chaîne</em>. La valeur de l’enregistrement à stocker.
            </p>
</td>
</tr>
<tr>
<td>`options`</td>
<td>
<p>
<em>Objet facultatif</em>. Options permettant de régler le comportement du cache.
            </p>
</td>
</tr>
<tr>
<td>`options.expires_at`</td>
<td>
<p>
              Numéro facultatif. L’heure d’expiration absolue en millisecondes depuis l’époque Unix. Bien que les enregistrements mis en cache puissent être expulsés plus tôt, ils ne resteront jamais au-delà de la valeur `expires_at` fournie.
            </p>
<p>
<b>Remarque :</b> Cette valeur ne doit pas être fournie si une valeur a également été donnée pour `ttl`. La première des deux échéances sera retenue si les deux options sont fournies.
            </p>
</td>
</tr>
<tr></tr>
</tbody>
</table>

### `api.notification`

Indique si l’événement notifié devrait être réessayé ou abandonné.
Consultez chacune de ces méthodes pour obtenir plus de détails sur le comportement réel.
Si plusieurs appels sont effectués, seul le dernier événement de notification sera pris en compte.

### `api.notification.drop(reason)`

Lorsqu’il est appelé, l’événement de notification est considéré comme un échec sans possibilité de récupération.

Nous enregistrerons une erreur pour cet événement [`(fn)`](https://auth0.com/docs/deploy-monitor/logs/log-event-type-codes), mais nous ne l’enverrons plus à l’action à l’avenir.

Si vous avez besoin de réessayer cet événement de notification, envisagez plutôt d’appeler `api.notification.retry`.

<table class="table">
<thead>
<tr>
<th>Paramètre</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>`reason`</td>
<td>
<p>
<em>Chaîne facultative</em>. Cela sera inclus dans l’entrée du journal et vous aidera à analyser l’erreur plus en profondeur. Veuillez noter que ce champ se limite à 1 024 caractères et sera tronqué s’il est plus long.
            </p>
</td>
</tr>
</tbody>
</table>

### `api.notification.retry(reason)`

Lorsqu’il est appelé, l’événement de notification est considéré comme un échec avec possibilité de récupération.

Nous enregistrerons une erreur pour cet événement [`(fn)`](https://auth0.com/docs/deploy-monitor/logs/log-event-type-codes), mais nous le réessaierons jusqu’à cinq fois dans les prochaines minutes.

Si vous estimez que cet événement de notification ne doit pas être réessayé, envisagez plutôt d’appeler `api.notification.drop`.

<table class="table">
<thead>
<tr>
<th>Paramètre</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>`reason`</td>
<td>
<p>
<em>Chaîne facultative</em>. Cela sera inclus dans l’entrée du journal et vous aidera à analyser l’erreur plus en profondeur. Veuillez noter que ce champ se limite à 1 024 caractères et sera tronqué s’il est plus long.
            </p>
</td>
</tr>
</tbody>
</table>